samples/subsys/ipc/ipc_service/static_vrings/remote/src/main.c

```diff
@@ -181,16 +181,28 @@ K_THREAD_DEFINE(ipc0B_thread_id, STACKSIZE, ipc0B_entry, NULL, NULL, NULL, PRIOR
 
 /*
  * ==> THREAD 1 (IPC instance 1) <==
+ *
+ * NOTE: This instance is using the NOCOPY copability of the backend.
  */
 
+static struct ipc_ept ipc1_ept;
+static void *recv_data;
+
 static void ipc1_ept_bound(void *priv)
 {
 	k_sem_give(&ipc1_bound_sem);
 }
 
 static void ipc1_ept_recv(const void *data, size_t len, void *priv)
 {
-	ipc1_received_data = *((uint8_t *) data);
+	int ret;
+
+	ret = ipc_service_hold_rx_buffer(&ipc1_ept, (void *) data);
+	if (ret < 0) {
+		printk("ipc_service_hold_rx_buffer failed with ret %d\n", ret);
+	}
+
+	recv_data = (void *) data;
```

Comment:
Here if two call back are received before you release the RX buffer the recv_data will be overwritten.
I suggest to add a comment explaining that the example is designed to support only a synchronous
"request-answer" mechanism. for asynchronous a chain list should be implemented.

Reply:
ok